.github/workflows/eslint-prettier-on-pr.yml

```diff
@@ -0,0 +1,31 @@
+# workflow to enforce style and linting on PR
+name: eslint-prettier
+on:
+  pull_request:
+    branches:
+      - main
+      - develop
+jobs:
+  format-and-lint:
+    runs-on: ubuntu-latest
+    permissions:
+      # Give the default GITHUB_TOKEN write permission to commit and push the changed files back to the repository.
+      contents: write
+      
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+      # Setup npm environment to run tests
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '16.x'
```

Comment:
Why version 16? 18 is the current LTS and 20 is the current version.

Reply:
I'll set it to 18.x